serde/src/java/org/apache/hadoop/hive/serde2/OpenCSVSerde.java

```diff
@@ -178,19 +185,26 @@ public Object deserialize(final Writable blob) throws SerDeException {
   private CSVReader newReader(final Reader reader, char separator, char quote, char escape) {
     // CSVReader will throw an exception if any of separator, quote, or escape is the same, but
     // the CSV format specifies that the escape character and quote char are the same... very weird
+    CSVParser parser;
     if (CSVWriter.DEFAULT_ESCAPE_CHARACTER == escape) {
-      return new CSVReader(reader, separator, quote);
+      parser = new CSVParserBuilder()
+          .withSeparator(separator)
+          .withQuoteChar(quote)
+          .build();
     } else {
-      return new CSVReader(reader, separator, quote, escape);
+      parser = new CSVParserBuilder()
```

Comment:
can we avoid duplication by creating a builder var or using ternary operator? the only diff here is the missing escape char setup